akka-projection-core/src/main/scala/akka/projection/TelemetrySpi.scala

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (C) 2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.projection
+
+import scala.collection.immutable
+
+import akka.actor.typed.ActorSystem
+import akka.annotation.InternalApi
+import akka.annotation.InternalStableApi
+
+/**
+ * Service Provider Interface (SPI) for collecting metrics from projections.
+ *
+ * Implementations must include a single constructor with two arguments: [[ProjectionId]]
+ * and [[ActorSystem[_]].  To setup your implementation, add a setting on your `application.conf`:
+ *
+ * {{{
+ * akka.projection.telemetry.fqcn = com.example.MyMetrics
+ * }}}
+ */
+trait Telemetry {
```

Comment:
Do we have insights to decide if we need both StatusObserver and Telemetry now?
The corresponding methods are called from the same points, with some small differences that we can align.

One difference is that we create a new Telemetry instance for each ProjectionId, while StatusObserver is passing in the projectionId in all methods. How are the metrics collectors in Cinnamon designed? Are they anyway based on Map with ProjectionId as key or is there an advantage of this design with a new instance for each ProjectionId?

We could change StatusObserver to follow this pattern instead if we think it's better.